airbyte-protocol/protocol-models/src/main/resources/airbyte_protocol/airbyte_protocol.yaml

```diff
@@ -201,6 +205,31 @@ definitions:
         enum:
           - system_error
           - config_error
+  AirbyteEstimateTraceMessage:
+    type: object
+    additionalProperties: true
+    required:
+      - name
+      - type
+    properties:
+      name:
+        description: The name of the stream
+        type: string
+      type:
+        description: The type of estimate
+        type: string
+        enum:
+          - STREAM
+          - SYNC
```

Comment:
Perhaps the question is about `type` rather than `level`

Ideally, the source reports the estimates per-stream (`type=stream`):
* stream "users" has 100 records @ 1000 bytes
* stream "purchases" has 50 records @ 300 bytes

But some sources, CDC and GLOBAL state API sources come to mind, can only produce an estimate for the entirety of the sync, not per stream (`type-sync`). 
* this sync will produce 150 records @ 1300 bytes

This means in the UI we can show a total sync progress bar, but won't be able to show the per-stream break-out information... which is still far better than nothing!